File: clients/rs/src/features/wrap_type_bignumber.rs
use std::error::Error;
use polywrap_client::{core::uri::Uri, client::PolywrapClient};
use polywrap_client_default_config::SystemClientConfig;
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Serialize, Deserialize)]
struct InnerArgObj {
  prop1: String
}

#[derive(Serialize, Deserialize)]
struct Args {
  arg1: String,
  obj: InnerArgObj
}

#[derive(Serialize, Deserialize)]
struct InputObj {
  args: Args,
}

pub fn run_test_case(input: &Value) -> Result<(), Box<dyn Error>> {
  let input_obj: InputObj = serde_json::from_value(input.clone())?;
  let args = input_obj.args;

  let binding = std::env::current_dir()?.join("../../wraps");
  let root = binding.to_str().unwrap();
  let uri: Uri = format!("fs/{root}/bignumber-type/implementations/as").try_into().unwrap();

  let client: PolywrapClient = PolywrapClient::new(SystemClientConfig::default().into());

  println!("Invoking method");

  let result = client.invoke::<Value>(
    &uri,
    "method",
    Some(&polywrap_client::msgpack::to_vec(&args)?),
    None,
    None
  );

  match result {
    Ok(result) => {
      let result = result.as_str().unwrap();
      println!("Result: {result}");
      println!("Success!");
    },
    Err(err) => panic!("{err}"),
  }
  
  Ok(())
}

File: clients/swift/Sources/Readiness/Features/WrapTypeBignumber.swift
import PolywrapClient
import Foundation

struct WrapTypeBignumberTest: Feature {
    func runTestCase(input: Any) throws -> Void {
        guard let inputObj = input as? [AnyHashable: Any] else {
           throw InputError.expectedObject
        }

        guard let argsDict = inputObj["args"] as? [AnyHashable: Any],
            let arg1 = argsDict["arg1"] as? String,
            let obj = argsDict["obj"] as? [AnyHashable: Any],
            let prop1 = obj["prop1"] as? String else {
            throw InputError.expectedObject
        }
        guard let resourcePath = Bundle.main.resourcePath else {
            fatalError("Resource folder not found")
        }
        let root = resourcePath + "/Readiness_Readiness.bundle/Contents/Resources"

        let uri = try Uri("file/\(root)/wraps/bignumber-type/implementations/as")

        let builder = BuilderConfig().addSystemDefault()
        let client = builder.build()

        print("Invoking method")
        let result: String = try client.invoke(uri: uri, method: "method", args: Args(arg1, Obj(prop1)), env: nil)
        print("Result: \(result)")
        print("Success!")
    }
}
